src/android/toga_android/app.py

```diff
@@ -92,3 +112,19 @@ def set_on_exit(self, value):
 
     def add_background_task(self, handler):
         self.loop.call_soon(wrapped_handler(self, handler), self)
+
+    async def invoke_intent_for_result(self, intent):
+        """
+        Calls an Intent and waits for its result
+
+        :param Intent intent: The Intent to call
+        :returns: A Dictionary containing "resultCode" (int) and "resultData" (Intent or None)
+        :rtype: dict
+        """
+        self._listener.last_intent_requestcode += 1
+        code = self._listener.last_intent_requestcode
```

Comment:
This is a great use of counters to avoid object leaks, and excellent bookkeeping of the `running_intents` dict.